README.md

```diff
@@ -32,7 +35,94 @@ Since the project is under active development, we recommend watching the reposit
+
+* Inventory setting. There is an explicit call to set the inventory. This will update the internal
+inventory of the client and be reported in the next inventory interval.
+
+See the source code for more.
+
+
+### Update Modules API
+
+Mender MCU models the Update Modules API that we have been using in the regular Mender client for
+years. Refer to the [Update Modules chapter in Mender
+Docs](https://docs.mender.io/artifact-creation/create-a-custom-update-module) for an introduction.
+In Mender MCU, the modules are not "executables" but rather a piece of code that is integrated with
+the client to handle each type of update.
+
+#### Update Modules State Machine
+
+As stated above, an Update Module for Mender MCU is just C code. In particular, it is defined by a C
+struct in [mender-zephyr-image-update-module.h](include/mender-zephyr-image-update-module.h). The
+most important part of the struct is the array of callbacks, one to be called for each state.
+
+See [the state machine workflow
+diagram](https://docs.mender.io/artifact-creation/create-a-custom-update-module#the-state-machine-workflow)
+to learn about the flow between each state. An Update Module does not need to implement all of them,
+only the ones that are relevant for a particular type of update.
+
+We provide `zephyr-image` Update Module, which implements the update process for a Zephyr OS update
+integrated with MCUboot bootloader. It's [source code](core/src/mender-zephyr-image-update-module.c)
+can be inspected for inspiration and better understanding of the expected behavior of each state.
+
+After writing the code, you need to register the Update Module into the Mender MCU. See Register
+update module function in [`mender-client.h`](include/mender-client.h).
+
+
+## Known issues
+
+There are a number of outstanding known issues with the current code:
+
+* No proper scheduler. Internal ticket at [MEN-7536](https://northerntech.atlassian.net/browse/MEN-7536).
+  So far we wrote a minimal scheduling single threaded logic, that only allow for one operation at a
+  time. The client run gets scheduled every polling interval and would conduct the full update (if
+  any) at once. We plan to redesign it better to allow for re-entry and resource protection, so that
+  we could have inventory being reported independently, for example.
+
+* Inventory feature disabled.
+  Related to the above, there is no inventory reporting at the moment.
+
+* Freeze during a deployment. Internal ticket at MEN-TODO](https://northerntech.atlassian.net/browse/MEN-TODO).
+  We are experiencing that the client can freeze during context switch, when attending a kernel's
+  ISR, for example. This can happen during the download of the update.
+
+* Logging disabled.
+  Related to the above, as a temporary mitigation we have disabled logging from the client which
+  makes the risk of these freezes very small, although not completely gone.
```

Comment:
No longer true.

Reply:
What do you mean? It is disabled in the "last stable". I'll leave it.

Reply:
Is logging still disabled? I thought it wasn't. I guess it's just no an `#error` anymore?

Reply:
[It is](https://github.com/mendersoftware/mender-mcu/blob/v0.2-update-modules/platform/log/zephyr/src/mender-log.c#L21).